### DISCUSSION TITLE
Performance measurement of `TemporalTransformManager`

### DISCUSSION THREAD

**Discussion by kopytjuk** [dfki-ric/pytransform3d#312] [Show and tell]
A small profiling measurement, of how fast the new vectorized `TemporalTransformManager` became, e.g. for 100000 timesteps, it just needs 3.8s on my MacBook M1 machine. Compared to that, a `for`-loop approach of each query timestep (i.e. the way to do that prior to the https://github.com/dfki-ric/pytransform3d/pull/301(, takes very long, so after 60s I just stopped the benchmark...

Script:

```python
# profile_transform.py

import numpy as np

from pytransform3d.test.test_transform_manager import create_sinusoidal_movement
from pytransform3d.transform_manager import (
    NumpyTimeseriesTransform,
    TemporalTransformManager,
)

if __name__ == "__main__":
    # create entities A and B together with their transformations from world
    duration = 10.0  # [s]
    sample_period = 0.01  # [s]
    velocity_x = 1  # [m/s]
    time_A, pq_arr_A = create_sinusoidal_movement(
        duration, sample_period, velocity_x, y_start_offset=0.0, start_time=0.1
    )
    A2world = NumpyTimeseriesTransform(time_A, pq_arr_A)

    time_B, pq_arr_B = create_sinusoidal_movement(
        duration, sample_period, velocity_x, y_start_offset=2.0,
        start_time=0.35
    )
    B2world = NumpyTimeseriesTransform(time_B, pq_arr_B)

    tm = TemporalTransformManager()

    tm.add_transform("A", "W", A2world)
    tm.add_transform("B", "W", B2world)

    N = 1000000
    query_times = np.random.uniform(0.5, 9.0, N)
    A2B_at_query_time = tm.get_transform_at_time("A", "B", query_times)
```

Results:

with: `pyinstrument -r html -o result.html profile_transform.py`

![image](https://github.com/user-attachments/assets/0130541d-99eb-478a-87fc-2f0a9f147e55)

Kudos to @JonasHablitzel and @AlexanderFabisch

**Comment by AlexanderFabisch** [comment]
Then your laptop is faster than my desktop computer. :smiling_face_with_tear: 

5.3 s on i7-10700KF CPU @ 3.80GHz / 8 cores.

Anyway, great work @JonasHablitzel !
